android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -892,14 +892,13 @@ public void loadNewContent() {
 
     @Override
     public void getFeed(boolean isNewContent) {
-        PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK, () -> {
-            if (!isNewContent) {
-                mNtpAdapter.setImageCreditAlpha(1f);
-                mNtpAdapter.setNewsLoading(true);
-            }
-            initBraveNewsController();
-            mBraveNewsController.getFeed(feed -> { runFeed(isNewContent, feed); });
-        });
+        if (!isNewContent) {
+            mNtpAdapter.setImageCreditAlpha(1f);
+            mNtpAdapter.setNewsLoading(true);
+        }
+        initBraveNewsController();
+        PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK,
+                () -> { mBraveNewsController.getFeed(feed -> { runFeed(isNewContent, feed); }); });
     }
 
     private void runFeed(boolean isNewContent, Feed feed) {
```

Comment:
it looks like `processFeed` inside has an access to `mNtpAdapter` adapter as well.

Reply:
processFeed inside all the code is running in ui thread